### PR TITLE
Pass flavor in CREATE MODEL

### LINKED ISSUE #215: Add a flavor clause in CREATE MODEL
We can specify the model flavor , similar to what mlflow does.

```sql
CREATE MODEL model_name
FLAVOR [tensorflow|pytorch|xgboost]
OPTIONS (k=v, k=v, ...)
USING "model_uri"
```

### PR DESCRIPTION
**Pull request by eddyxu**
This closes #215

### CODE REVIEW

File: src/main/scala/ai/eto/rikai/sql/model/ModelSpec.scala [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+package ai.eto.rikai.sql.model
+
+import scala.collection.JavaConverters.mapAsJavaMap
+
+/** Model Spec is used to pass the create model information to
+  * python ModelRegistry.
+  */
+class ModelSpec(
+    val name: Option[String],
+    val uri: String,
```
Comment (changhiskhan) [inline src/main/scala/ai/eto/rikai/sql/model/ModelSpec.scala:26]: nit: let's call this spec_uri as well to be consistent with the new python code?

File: src/main/scala/ai/eto/rikai/sql/model/Model.scala [hunk 2]
```diff
@@ -68,10 +70,18 @@ class ModelNameException(message: String) extends Exception(message);
   * @param uri the model uri.
   * @param funcName the name of a UDF which will be called when this model is invoked.
   */
-class SparkUDFModel(val name: String, val uri: String, val funcName: String)
-    extends Model
+class SparkUDFModel(
+    val name: String,
+    val uri: String,
```
Comment (changhiskhan) [inline src/main/scala/ai/eto/rikai/sql/model/Model.scala:75]: i forgot is this is the artifact or the spec uri

**Review by changhiskhan — APPROVED** [review]

File: src/main/scala/ai/eto/rikai/sql/model/Model.scala (hunk 2 above)
Comment (eddyxu) [reply src/main/scala/ai/eto/rikai/sql/model/Model.scala:75]: This is the URI specified from `CREATE TABLE using URI`?

File: src/main/scala/ai/eto/rikai/sql/model/ModelSpec.scala (hunk 1 above)
Comment (eddyxu) [reply src/main/scala/ai/eto/rikai/sql/model/ModelSpec.scala:26]: sure

**Pull request merged**